### PR TITLE
Fixes for Issue #936

### LINKED ISSUE #936: Fails to make a universe from 2Y2N.pdb
When a universe should be created, a KeyError is raised.

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-505-4af8f1cb9d1b> in <module>()
----> 1 mda.Universe(path + '2Y2N.pdb')

/usr/local/lib/python2.7/dist-packages/MDAnalysis/core/AtomGroup.pyc in __init__(self, *args, **kwargs)
   4337         try:
   4338             with parser(self.filename, universe=self) as p:
-> 4339                 self._topology = p.parse()
   4340         except IOError as err:
   4341             raise IOError("Failed to load from the topology file {0}"

/usr/local/lib/python2.7/dist-packages/MDAnalysis/topology/PDBParser.pyc in parse(self)
     88         structure['atoms'] = atoms
     89 
---> 90         bonds = self._parsebonds(atoms)
     91         structure['bonds'] = bonds
     92 

/usr/local/lib/python2.7/dist-packages/MDAnalysis/topology/PDBParser.pyc in _parsebonds(self, atoms)
    181                 atom, atoms = _parse_conect(line.strip())
    182                 for a in atoms:
--> 183                     bond = tuple([mapping[atom], mapping[a]])
    184                     bonds.add(bond)
    185 

KeyError: 3635

``` python
import MDAnalysis as mda

u = mda.Universe('2Y2N.pdb')
```

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #936

Changes made in this Pull Request:
 - Conect records to TER ignored

PR Checklist
------------
 - [x] Tests?
 - [x] Docs? N/A
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
Also fixes #937 now too!

**Comment by jbarnoud** [comment]
Looks good to me. Would it be possible to issue a warning, though?

**Comment by richardjgowers** [comment]
@jbarnoud I did think about it, but opening normal files from the PDB shouldn't give warnings, I thought that should be reserved for PDB files that are non-standard/weird

**Comment by jbarnoud** [comment]
@richardjgowers I think it is dangerous to assume that PDB files from the PDB are nice and clean. I've seen results of high throughput study on PDB files change completely once you fix PDB oddities. If I work on an oddly formated files, I'd rather know it, especially if it comes from the PDB that I am supposed to trust.

**Comment by richardjgowers** [comment]
@jbarnoud ahh ok, I don't do biothings, I assumed it was considered gospel.  I'll add a few warnings then

**Comment by richardjgowers** [comment]
@jbarnoud over to you

File: package/MDAnalysis/topology/PDBParser.py [hunk 1]
```diff
@@ -96,64 +96,63 @@ def _parseatoms(self):
...
-
-                    atom = Atom(iatom, name, atomtype, resName, resid,
-                                segid, mass, charge,
-                                bfactor=tempFactor, serial=serial,
-                                altLoc=altLoc, universe=self._u,
-                                resnum=resSeq)
-                    iatom += 1
-                    atoms.append(atom)
+        resid_prev = 0  # resid looping hack
+        for line in anyopen(self.filename, 'rt'):
```
Comment (jbarnoud) [inline package/MDAnalysis/topology/PDBParser.py]: Why not using `with`?

Comment (richardjgowers) [inline package/MDAnalysis/topology/PDBParser.py]: It was an extra level of indentation we didn't need

Comment (richardjgowers) [inline package/MDAnalysis/topology/PDBParser.py]: Ah, but maybe the `with` was more safe after reading about it.. I'll revert this

**Pull request merged**